### PR TITLE
genfit: Define new version, update dependency

### PR DESCRIPTION
**Pull request by plexoos**
- Add dependency on eigen package
- Add last version known to work with root <6.18

  The GenFit project lacks any release or tag versions, therefore, we are forced
  to use date versions for now

- Update title wording and default archive type (tar.gz compression prevails over zip)

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
@mirguest Waiting for your approval to merge. Code LGTM.

**Comment by mirguest** [comment]
Hi @plexoos ,
There is a new release in GenFit now: https://github.com/GenFit/GenFit/releases/tag/02-00-00
Would you like to add this? Otherwise I could create another PR. 
Thanks.

Tao

**Review by mirguest — DISMISSED** [review]
Looks good. Thank you!
Tao

File: var/spack/repos/builtin/packages/genfit/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/genfit/package.py]: @alalazo shouldn't this be:
```suggestion
    conflicts('^root@6.18.00:', when='@b496504a')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by adamjstewart** [comment]
Can you resolve the merge conflict?

File: var/spack/repos/builtin/packages/genfit/package.py
Comment (alalazo) [reply var/spack/repos/builtin/packages/genfit/package.py]: I think we shouldn't be using a conflict in this case, but a normal depends on.

File: var/spack/repos/builtin/packages/genfit/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/genfit/package.py:28]: At least with the new concretizer you should be able to model this as:
```suggestion
    depends_on('eigen')
    depends_on('root@:6.17.99', when='@b496504a')
```
and you can thus remove the conflict directive below.

Comment (plexoos) [reply var/spack/repos/builtin/packages/genfit/package.py:28]: If this works it is fine by me

**Comment by alalazo** [comment]
@plexoos

**Comment by plexoos** [comment]
Please merge if looks good

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
